concepts/smart-pointers/introduction.md

```diff
@@ -0,0 +1,175 @@
+# Introduction
+
+Smart pointers are a modern C++ feature designed to provide automatic memory management, helping to prevent memory leaks and dangling pointers commonly associated with raw pointers.
+They act as wrappers around raw pointers, adding additional functionality such as automatic memory deallocation when the pointer is no longer needed.
+
+## General Syntax
+
+Smart pointers are typically implemented as template classes in the C++ standard library.
+The two most commonly used smart pointers are `std::unique_ptr` and `std::shared_ptr`. 
+
+```cpp
+#include <memory>
+// Declaring a unique pointer
```

Comment:
Here and in the rest of the file: 
Yes, that's a a declaration but probably more importantly: it's also a definition.